### DISCUSSION TITLE
Access to stdout/stderr ?

### DISCUSSION THREAD

**Discussion by bstopp** [avajs/ava#3002] [Q&A] [answered]
Is there a way get `stdout` or `stderr` from the tests? I see some of my tests output, but not all. (i.e. spawned process' logs show in the middle of the test output, but the logs of the code running in the test thread do not show up.)

I'd like to be able to validate that the console logs contain certain messages.

For context I have to disabled `workerThreads` as I have to change the cwd.

TIA.

**Comment by novemberborn** [comment, marked as answer]
`stdout` and `stderr` from test workers are passed along to the AVA output. They both end up on either `stdout` or `stderr` (I can't recall). AVA does not provide a direct mechanism for observing them from inside your tests. It should be possible to modify these output streams and intercept the output that way.
